core/iwasm/fast-jit/fe/jit_emit_control.c

```diff
@@ -584,16 +584,9 @@ jit_compile_op_block(JitCompContext *cc, uint8 **p_frame_ip,
             SET_BB_END_BCIP(cc->cur_basic_block, *p_frame_ip - 1);
             SET_BB_BEGIN_BCIP(block->basic_block_entry, *p_frame_ip);
 
-            if (else_addr) {
-                if (!push_jit_block_to_stack_and_pass_params(
-                        cc, block, block->basic_block_entry, value))
-                    goto fail;
-            }
-            else {
-                if (!push_jit_block_to_stack_and_pass_params(
-                        cc, block, block->basic_block_else, value))
-                    goto fail;
-            }
+            if (!push_jit_block_to_stack_and_pass_params(
+                    cc, block, block->basic_block_entry, value))
```

Comment:
Should also modify L195, change `if (basic_block == block->basic_block_entry)` to:
`if (block->wasm_code_else)`, if has else branch, we should generate `BNE cond, basic_block_entry, basic_block_else`, if no, we should generate `BNE cond, basic_block_entry, basic_block_end`?

Reply:
That is the part I am not sure about it yet. Let me take more time to dig into it.